aws-http-nio-client-netty/src/main/java/software/amazon/awssdk/http/nio/netty/EventLoopGroupFactory.java

```diff
@@ -16,130 +16,17 @@
 package software.amazon.awssdk.http.nio.netty;
 
 import io.netty.channel.EventLoopGroup;
-import io.netty.channel.epoll.Epoll;
-import io.netty.channel.epoll.EpollEventLoopGroup;
-import io.netty.channel.nio.NioEventLoopGroup;
-import java.util.Optional;
-import java.util.concurrent.ThreadFactory;
-import software.amazon.awssdk.utils.builder.CopyableBuilder;
-import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
 
 /**
- * Configurable factory for creating {@link EventLoopGroup} instances. Will choose the optimal implementation of
- * {@link EventLoopGroup} per the platform (i.e. if on Linux then {@link EpollEventLoopGroup} will be used).
+ * Factory interface for creating an {@link EventLoopGroup}.
+ *
+ * @see DefaultEventLoopGroupFactory
  */
-public final class EventLoopGroupFactory
-        implements ToCopyableBuilder<EventLoopGroupFactory.Builder, EventLoopGroupFactory> {
-
-    private final Integer numberOfThreads;
-    private final ThreadFactory threadFactory;
-
-    private EventLoopGroupFactory(DefaultBuilder builder) {
-        this.numberOfThreads = builder.numberOfThreads;
-        this.threadFactory = builder.threadFactory;
-    }
-
-    /**
-     * @return The number of threads currently configured for the {@link EventLoopGroup} or an empty {@link Optional} if not set.
-     */
-    public Optional<Integer> numberOfThreads() {
-        return Optional.ofNullable(numberOfThreads);
-    }
-
-    /**
-     * @return The currently configured {@link ThreadFactory} or an empty {@link Optional} if not set.
-     */
-    public Optional<ThreadFactory> threadFactory() {
-        return Optional.ofNullable(threadFactory);
-    }
+public interface EventLoopGroupFactory {
```

Comment:
I'd definitely put the migration of this to an interface in #49 - it's going to make for smaller diffs for both PRs